Add command to scaffold, download input and read challenge for current date

tomvaneyck opened the pull request:
This pull request combines the scaffold, download and read commands into one, auto selecting the current date or accepting an optional day.

Command:
```sh
cargo today [<day>]
```

tomvaneyck commented:
It seems like I had the same idea as someone else a moment ago (#42) 😃. I leave the choice on the final implementation to you.

fspoettel commented:
Hey, thanks for the PR! I like the idea of a `today` command. I'll review this PR in detail when I have some time in the next couple of days.

I'm not 100% sure if `today` should have the `day` flag or if we rather want to split it into `cargo scaffold [-d, --download] <arbitrary_day>` and `cargo today` (magical shorthand usable in december).

fspoettel commented on README.md:
````diff
@@ -160,6 +160,32 @@ cargo read <day>
+
+```sh
+# example: `cargo today` on December 1st
+cargo today [<day>]
````
We added the `--download` flag to `solve` in the mean time, so we should remove it here. I think that having a clear separation makes the most sense - have `solve` work anytime and `today` be automatic in december.

fspoettel commented on Cargo.toml:
```diff
@@ -14,4 +14,5 @@ doctest = false
 test_lib = []
 
 [dependencies]
+chrono = "0.4.31"
```
I'm a bit reluctant to add dependencies on the template-level, mainly because we currently can't compile them in advance and the `cargo` commands hide the template compilation output.

Task for myself: need to check `chrono`'s overhead and see if we need to do something about it.

tomvaneyck commented:
I think I addressed all comments, except for the one about using `chrono`. However, using that crate seems like the only way to get the current day of the month.

fspoettel commented:
What we can definitely do is put `chrono` behind a feature flag and add it to the `today` alias.

fspoettel commented:
Thank you! ❤️
